### ISSUE TITLE
Separate bug fix from feature (git)

### ISSUE THREAD

**Issue by laurent-laporte-pro** [c0fec0de/anytree#117]
It would be useful to have a "2.7" git branch to track bug fixes on the 2.7.x version and use the "master" branch for next minor release (a.k.a. 2.8.x version).

What is the role of the "develop" branch. Currently "develop" == "master" ?

**Comment by c0fec0de** [comment]
Anytree is released according to pep440 (https://www.python.org/dev/peps/pep-0440/)

We use the simple major.minor.patch system. Bugfixes increase the patch number. New features increase the minor number. API breaks increase the major.

As soon as there are new features or bug fixes i intend to just increase the corresponding minor or patch number. From the current perspective I would only maintain an extra branch if there is an API break.

Any concerns about that approach?

**Comment by laurent-laporte-pro** [comment]
All right, I understand that you use a [semantic versioning](https://semver.org/). It wasn’t the purpose of my question. So, I’m trying to be clearer:

Well, it wasn't easy to me to understand which branch to use for my contributions (PRs). I can, for instance, have a bug fix or a feature:
* If I have a bug fix, I want to work on the **current** minor version (2.7.x in your project). My commits should be in the current production branch, deemed reliable. The next version number should be 2.7.x+1. If relevant, my commit should be cherry-picked to **next** minor version branch.
* If I have a new feature, I want to work on the **next** minor version (2.7.0 in your project). My commits should be in the current development branch (not yet stable).

Here is what I use for my projects (personal or professional):
* The **master** branch is used for the current development branch (not yet stable).
* The **X.Y** branch is used for the current production branch. I can maintain several productions branches, but I often keep only one (the last minor release).

In summary, if I want to fix a bug, I work on the “X.Y” branch (and cherry-pick on “master” if relevant), and I work on the “master” branch for new features.

What I am suggesting to you is to rename your “master” branch to “2.7” and to rename your “develop” branch to “master”.
Or, you can add a CONTRIBUTING.rst file to explain what the contributors should do? Should I create a PRs for that?

**Comment by c0fec0de** [comment]
I know this approach. Just did not use it here. But as this project becomes more and more used, let's start with a CONTRIBUTING.rst. Feel free to create one.
We will start with a *X.Y* branch with 2.8.

* New features should base on master
* Bug fixes should be done on the X.Y branch
